src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs

```diff
@@ -232,6 +234,124 @@ public JsonNode Root
             }
         }
 
+        /// <summary>
+        /// Creates a new instance of the <see cref="JsonNode"/>. All children nodes are recursively cloned.
+        /// </summary>
+        public JsonNode DeepClone()
+        {
+            if (this is JsonObject jObject)
```

Comment:
Couldn't all these checks be replaced with an internal virtual method like you did in DeepEquals?